Dockerfile

```diff
@@ -1,151 +1,114 @@
-	npm_install="${NPM_VERSION}" npm_config_prefix=/tmp sh ./install.sh && \
-	JOBS=MAX /tmp/bin/npm ci --no-optional --unsafe-perm --production
+# Install node from baleanas prebuilt cache
+RUN curl -SLO "${NODE_LOCATION}" \
+	&& curl  -SLO "${S3_BASE}/${SUMS}" \
+	&& grep "${NODE_ARCHIVE}" ${SUMS} | sha256sum -c - \
+	&& tar -xzf "${NODE_ARCHIVE}" -C /usr/local --strip-components=1 \
+	&& rm -f "${NODE_ARCHIVE}" \
+	&& strip /usr/local/bin/node
 
-# Remove various uneeded filetypes in order to reduce space
-# We also remove the spurious node.dtps, see https://github.com/mapbox/node-sqlite3/issues/861
-RUN find . -path '*/coverage/*' -o -path '*/test/*' -o -path '*/.nyc_output/*' \
-	-o -name '*.tar.*'      -o -name '*.in'     -o -name '*.cc' \
-	-o -name '*.c'          -o -name '*.coffee' -o -name '*.eslintrc' \
-	-o -name '*.h'          -o -name '*.html'   -o -name '*.markdown' \
-	-o -name '*.md'         -o -name '*.patch'  -o -name '*.png' \
-	-o -name '*.yml'        -o -name "*.ts" \
-	-delete \
-	&& find . -type f -path '*/node_modules/sqlite3/deps*' -delete \
-	&& find . -type f -path '*/node_modules/knex/build*' -delete \
-	&& rm -rf node_modules/sqlite3/node.dtps
+COPY package*.json ./
 
-COPY entry.sh package.json rootfs-overlay/usr/src/app/
+RUN npm ci
 
-RUN rsync -a --delete node_modules rootfs-overlay /build
+COPY webpack.config.js fix-jsonstream.js hardcode-migrations.js tsconfig.json tsconfig.release.json ./
+COPY src ./src
+COPY test ./test
+COPY typings ./typings
 
-RUN [ "cross-build-end" ]
-
-##############################################################################
+RUN npm run test-nolint \
+	&& npm run build
 
-# Minimal runtime image
-FROM balena/$ARCH-supervisor-base:v1.4.7
+# Run the production install here, to avoid the npm dependency on
+# the later stage
+RUN npm ci --production --no-optional --unsafe-perm \
+	&& npm cache clean --force \
+	# For some reason this doesn't get cleared with the other
+	# cache
+	&& rm -rf node_modules/.cache \
+	# Remove various uneeded filetypes in order to reduce space
+	# We also remove the spurious node.dtps, see https://github.com/mapbox/node-sqlite3/issues/861
+		&& find . -path '*/coverage/*' -o -path '*/test/*' -o -path '*/.nyc_output/*' \
+			-o -name '*.tar.*'      -o -name '*.in'     -o -name '*.cc' \
+			-o -name '*.c'          -o -name '*.coffee' -o -name '*.eslintrc' \
+			-o -name '*.h'          -o -name '*.html'   -o -name '*.markdown' \
+			-o -name '*.md'         -o -name '*.patch'  -o -name '*.png' \
+			-o -name '*.yml'        -o -name "*.ts" \
+			-delete \
+			&& find . -type f -path '*/node_modules/sqlite3/deps*' -delete \
+		&& find . -type f -path '*/node_modules/knex/build*' -delete \
+		&& rm -rf node_modules/sqlite3/node.dtps
+
+
+RUN ["cross-build-end"]
+
+FROM balenalib/$ARCH-alpine-supervisor-base:3.11
+
+RUN ["cross-build-start"]
+
+RUN apk add --no-cache \
+	ca-certificates \
+	kmod \
+	iptables \
+	ip6tables \
+	rsync \
+	curl \
```

Comment:
Do you need curl specifically for something? If `wget` ships with alpine, can save the install.

Reply:
Nice, I didn't actually realise that wget came bundled as a busybox binary. I'll change the healthcheck to use this too.